sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -918,6 +919,31 @@ def undelete_blob(self, **kwargs):
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @distributed_trace()
+    def exists(self, **kwargs):
+        # type: (**Any) -> bool
+        """
+        Returns True if a blob exists with the defined parameters, and returns
+        False otherwise.
+
+        :param str version_id:
+            The version id parameter is an opaque DateTime
+            value that, when present, specifies the version of the blob to check if it exists.
+        :param int timeout:
+            The timeout parameter is expressed in seconds.
+        :returns: boolean
+        """
+        try:
+            blob_props = self._client.blob.get_properties(
+                snapshot=self.snapshot,
+                cls=deserialize_blob_properties,
+                **kwargs)
+            if blob_props and blob_props.is_current_version or blob_props and self.snapshot:
```

Comment:
this line is not necessary, that's why deserialize_blob_properties can be removed